couchbase/operations/management/query_index_drop.cxx

```diff
@@ -71,22 +72,28 @@ query_index_drop_request::make_response(error_context::http&& ctx, const encoded
                             index_not_found = true;
                         }
                         break;
+
                     case 12003: /* IKey: "datastore.couchbase.keyspace_not_found" */
                         bucket_not_found = true;
                         break;
+
                     case 12004: /* IKey: "datastore.couchbase.primary_idx_not_found" */
                     case 12016: /* IKey: "datastore.couchbase.index_not_found" */
                         index_not_found = true;
                         break;
+
+                    default:
+                        common_ec = management::extract_common_query_error_code(error.code, error.message);
+                        break;
                 }
                 response.errors.emplace_back(error);
             }
-            if (index_not_found) {
-                if (!ignore_if_does_not_exist) {
-                    response.ctx.ec = error::common_errc::index_not_found;
-                }
+            if (index_not_found && !ignore_if_does_not_exist) {
```

Comment:
This was written the way it was to prevent the internal_server_failure catch all in extract_common_error_code

Reply:
right, I should not change it